sdk/monitor/opentelemetry-exporter-azure-monitor/src/export/trace.ts

```diff
@@ -107,10 +107,10 @@ export class AzureMonitorTraceExporter implements SpanExporter {
     } catch (senderErr) {
       // Request failed -- always retry
       this._logger.error(
-        "Envelopes could not be exported and are not retriable. Error message:",
+        "Retrying due to transient client side error. Error message:",
         senderErr.message
       );
-      return ExportResult.FAILED_NOT_RETRYABLE;
+      return ExportResult.FAILED_RETRYABLE;
```

Comment:
Could this get into an endless loop if the processing always throws?

Reply:
Will loop until transport layer is able to communicate with Breeze endpoint, this code is triggered when the HTTP call throws an exception, if Breeze is down, machine trying to export doesn't have network connectivity, etc. When Breeze is reachable the response code will be send and handled by this exporter

Reply:
Hmm, is there a delay between retrying? Or a max retry count?
I know breeze doesn't support a "retry" count on the event, but just thinking about avoiding excessive CPU and looping etc

Reply:
That would be responsibility of the Span Processor
https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/sdk.md#span-processor

The exporter only tell the processor if it was able to export or not and if retry is needed

Reply:
Sorry got confused, we do take care about retrying we put events in local file and release them when SpanProcessor exports again and we get a 200

Reply:
so `core-http` itself will do some built-in retries that you can customize:

https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/policies/exponentialRetryPolicy.ts
https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/core/core-http/src/policies/systemErrorRetryPolicy.ts

I do wonder if we need to put some more checks into this handler. Is there any other way we could end up here besides the endpoint being unreachable?

Reply:
Retry policies are a good thing to have, we don't want to retry forever here and that is why we take the approach of storing the telemetry in disk and try to send the data later if these policies fail, I can see retry is in place for network errors and specific errors like internal server errors and timeouts,  maybe we can improve logic here and add similar checks like SystemErrorRetryPolicy and timeout check in ExponentialRetryPolicy, looking at the code I realize there could be issues with serialization and we will keep retrying with current code, I need to test this scenario better.

Reply:
I think I may have been being a little unclear, what I was getting at is this package uses a generated client here:

https://github.com/Azure/azure-sdk-for-js/blob/a60fab312ae0bdc3041aa2be9ed797dd668b97f5/sdk/monitor/opentelemetry-exporter-azure-monitor/src/platform/nodejs/httpSender.ts#L24

Which ultimately ends up (through its ServiceClient base class) using the default pipeline which automatically includes these policies:

https://github.com/Azure/azure-sdk-for-js/blob/a60fab312ae0bdc3041aa2be9ed797dd668b97f5/sdk/core/core-http/src/serviceClient.ts#L734

Reply:
These retry policies are not enough for us, if endpoint is unavailable the policies will retry a few times and forget about the request if I understand correctly, what I'm referring to is that to improve logic in exporter side to take similar checks as the policies and look for error and response codes to determine if request is retry able in our side

Reply:
Oh yeah I wasn't trying to say they were enough, just wanted you to know they were there. 👍 

I'm fine with this particular change if there's no concern about busy looping.

Reply:
Added network error check to avoid infinite looping for other kind of errors like serialization ones